syntax = "proto3";

package services;

// Represents a single frame of keypoints (e.g., 126 float values).
message KeypointFrame {
  repeated float keypoints = 1;
}

// NEW MESSAGE: Represents a complete sequence of frames for a single gesture.
message GestureSequence {
  repeated KeypointFrame frames = 1; // Renamed from 'sequence' to 'frames' for clarity
}

// =================== LSTM Service ===================
// Defines the prediction service that runs the LSTM model.
service LstmService {
  // A simple RPC call that takes multiple gesture sequences and returns predictions.
  rpc Predict(LstmRequest) returns (LstmResponse);
}

// The request message for the LstmService.
message LstmRequest {
  // Contains multiple independent gesture sequences.
  repeated GestureSequence gestures = 1; // Now 'gestures' is a repeated field of GestureSequence
}

// The response message from the LstmService.
message LstmResponse {
  string translated_text = 1;
}


// =================== NLP Service ===================
// Defines the text processing service that refines the prediction.
service NlpService {
  // A simple RPC call that takes raw text and returns a refined version.
  rpc Refine(NlpRequest) returns (NlpResponse);
}

// The request message for the NlpService.
message NlpRequest {
  string raw_text = 1;
}

// The response message from the NlpService.
message NlpResponse {
  string refined_text = 1;
}